src/message/renderMessages.js

```diff
@@ -2,49 +2,34 @@
 import React from 'react';
 
 import type { Message, Narrow } from '../types';
-import {
-  isTopicNarrow,
-  isPrivateOrGroupNarrow,
-  extractTypeToAndSubjectFromNarrow,
-} from '../utils/narrow';
+import { isTopicNarrow, isPrivateOrGroupNarrow } from '../utils/narrow';
 import MessageHeaderContainer from '../message/headers/MessageHeaderContainer';
 import MessageContainer from '../message/MessageContainer';
-import OutboxMessageContainer from '../message/OutboxMessageContainer';
 import TimeRow from '../message/TimeRow';
 import { isSameRecipient } from '../utils/message';
 import { isSameDay } from '../utils/date';
 
 export default (messages: Message[], narrow: Narrow) => {
   const list: Object[] = [];
   let prevItem;
+  const showHeader = !isPrivateOrGroupNarrow(narrow) && !isTopicNarrow(narrow);
 
   for (const item of messages) {
-    const isOutbox = !item.id;
+    const isOutbox = 'isOutbox' in item && item.isOutbox === true;
     const diffDays =
       prevItem && !isSameDay(new Date(prevItem.timestamp * 1000), new Date(item.timestamp * 1000));
     if (!prevItem || diffDays) {
       list.push(
         <TimeRow type="time_row" key={`time${item.timestamp}`} timestamp={item.timestamp} />,
       );
     }
-
-    const showHeader = !isPrivateOrGroupNarrow(narrow) && !isTopicNarrow(narrow);
     const diffRecipient = !isSameRecipient(prevItem, item);
-    const extractedOutboxData = isOutbox ? extractTypeToAndSubjectFromNarrow(item.narrow) : {};
     if (showHeader && diffRecipient) {
       list.push(
         <MessageHeaderContainer
           type="header"
-          key={`header${item.id}`}
-          message={
-            isOutbox
-              ? {
-                  ...item,
-                  display_recipient: extractedOutboxData[1],
-                  subject: extractedOutboxData[2],
-                }
-              : item
-          }
+          key={`header${isOutbox ? item.timestamp : item.id}`}
```

Comment:
Would it work if you just set the item.id to the timestamp in the selector? Then you don't need to have any special handling here.